contracts/jobs/libraries/JobLib.sol

```diff
@@ -51,8 +50,9 @@ library JobLib {
         }
 
         // Use block hash and block number of the block after a claim to determine if a segment
-        // should be verified
-        if (uint256(keccak256(_claimBlock, _claimBlockHash, _segmentNumber)) % _verificationRate == 0) {
+        // should be verified. Note: this check will only work if _claimBlock + 1 is within
+        // the last 256 blocks from the current block
+        if (uint256(keccak256(_claimBlock + 1, block.blockhash(_claimBlock + 1), _segmentNumber)) % _verificationRate == 0) {
```

Comment:
Should we add an `assert()` here to make sure that `_claimBlock + 1 > block.number - 256` ? I forget what block.blockhash returns if the requested block is outside the 256 range, but if it's 0 then this hash could still evaluate to something that `% verificationRate == 0`.

Reply:
Good call. `block.blockhash` returns 0x0 if the block number is outside the 256 range. The verification period + slashing period check upon initialization of the JobsManager should prevent `shouldVerifySegment` from ever being called with the value of claimBlock + 1 being outside the 256 block range (since it is only ever called within the verificationPeriod and slashingPeriod), but the assert would be useful to enforce/specify the invariant